### PR TITLE
[no-release-notes] doltcore/sqle/enginetest: Unskip some new format enginetests

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

File: go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go [hunk 1]
```diff
@@ -254,12 +242,8 @@ func TestScripts(t *testing.T) {
 		skipped = append(skipped,
 			// Different error output for primary key error
 			"failed statements data validation for INSERT, UPDATE",
-			// missing FK violation
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go:257]: There are duplicate skips in the TestScriptsPrepared method

File: go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go [hunk 2]
```diff
@@ -254,12 +242,8 @@ func TestScripts(t *testing.T) {
 		skipped = append(skipped,
 			// Different error output for primary key error
 			"failed statements data validation for INSERT, UPDATE",
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go:244]: Does your change below not address this discrepancy? Should be simple to make the two formats match their string output for this errror

**Review by zachmu — APPROVED** [review]
LGTM

File: go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go (hunk 2 above)
Comment (andrew-wm-arthur) [reply go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go:244]: there was another error that failed this test. will circle back

**Pull request merged**